src/domain/safe/safe.repository.ts

```diff
@@ -0,0 +1,35 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { Safe } from './entities/safe.entity';
+import { DefinedError, ValidateFunction } from 'ajv';
+import { JsonSchemaService } from '../schema/json-schema.service';
+import { safeSchema } from './entities/schemas/safe.schema';
+import { ITransactionApiManager } from '../interfaces/transaction-api.manager.interface';
+import { ValidationErrorFactory } from '../schema/validation-error-factory';
+import { ISafeRepository } from './safe.repository.interface';
+
+@Injectable()
+export class SafeRepository implements ISafeRepository {
+  private readonly isValidSafe: ValidateFunction<Safe>;
+
+  constructor(
+    @Inject(ITransactionApiManager)
+    private readonly transactionApiManager: ITransactionApiManager,
+    private readonly validationErrorFactory: ValidationErrorFactory,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.isValidSafe = jsonSchemaService.compile(safeSchema);
+  }
+
+  async getSafe(chainId: string, address: string): Promise<Safe> {
```

Comment:
Why chainId is a string?

Reply:
This was to be consistent to all the other `chainId` in the project. I can make a separate PR to change all of them